openhands/integrations/github/github_service.py

```diff
@@ -136,6 +139,134 @@ async def search_repositories(
 
         return repos
 
+    async def execute_graphql_query(self, query: str, variables: dict[str, Any]) -> dict[str, Any]:
+        """Execute a GraphQL query against the GitHub API."""
+        try:
+            async with httpx.AsyncClient() as client:
+                github_headers = await self._get_github_headers()
+                response = await client.post(
+                    f"{self.BASE_URL}/graphql",
+                    headers=github_headers,
+                    json={"query": query, "variables": variables}
+                )
+                response.raise_for_status()
+                
+                result = response.json()
+                if "errors" in result:
+                    raise GHUnknownException(f"GraphQL query error: {json.dumps(result['errors'])}")
+                
+                return result
+                
+        except httpx.HTTPStatusError as e:
+            if e.response.status_code == 401:
+                raise GhAuthenticationError('Invalid Github token')
+            raise GHUnknownException('Unknown error')
+
+        except httpx.HTTPError:
+            raise GHUnknownException('Unknown error')
+
+    async def get_suggested_tasks(self, repo_full_name: str) -> list[SuggestedTask]:
+        """
+        Get suggested tasks for a repository, including:
+        - PRs with merge conflicts
+        - PRs with failing GitHub Actions
+        - PRs with unresolved comments
+        - Issues with specific labels (help wanted, chore, documentation, good first issue)
+        """
+        query = """
+        query GetSuggestedTasks($owner: String!, $name: String!) {
+          repository(owner: $owner, name: $name) {
+            pullRequests(first: 100, states: [OPEN]) {
```

Comment:
we should filter for PRs that were opened by the authenticated user (if that's not already happening)

Reply:
@openhands please filter for PRs that were opened by the authenticated user